htsohm/htsohm.py

```diff
@@ -700,61 +770,11 @@ def worker_run_loop(run_id):
-                            all_accessed_bin_tuples = session \
-                                    .query(
-                                        Material.gas_adsorption_bin,
-                                        Material.surface_area_bin,
-                                        Material.void_fraction_bin) \
-                                    .filter(
-                                        Material.run_id == run_id,
-                                        Material.retest_passed != False,
-                                        Material.generation_index < config['children_per_generation']) \
-                                    .distinct().all()
-                            all_accessed_bins = [ [e[0], e[1], e[2]]
-                                    for e in all_accessed_bin_tuples]
-                            print_block('ANNEALING WITH MUTATION STRENGTH :\t{}' \
-                                    .format(config['initial_mutation_strength']))
-                            for some_bin in all_accessed_bins:
-                                print('Annealing bin :\t{}'.format(some_bin))
-                                mutation_strength = MutationStrength(run_id, gen + 1, *some_bin)
-                                mutation_strength.strength = config['annealing_strength']
-                                session.add(mutation_strength)
-
-                if config['interactive_mode'] == 'on':
-                    if material.generation_index == config['children_per_generation'] - 1 and gen > 0:
-                        parent_ids = get_all_parent_ids(run_id, gen)
-                        ms_bins = []
-                        for parent_id in parent_ids:
-                            parent_bin = session.query(Material).get(parent_id).bin
-                            if parent_bin not in ms_bins:
-                                ms_bins.append(parent_bin)
-                        print('The following bins contain parents for the preceding generation:')
-                        for some_bin in ms_bins:
-                            print('\t{}'.format(some_bin))
-                        input('Press Enter to begin calculating new mutation strength(s) :')
-                        counter = 1
-                        for some_bin in ms_bins:
-                            print('\nCalculating strength for bin :\t{}'.format(some_bin))
-                            print('Bin {} / {}'.format(counter, len(ms_bins)))
-                            calculate_mutation_strength(run_id, gen + 1, parent_bin)
-                            input('Press Enter to continue :')
+            # make sure worker just finished thev last material in its generation
+            if material.generation_index == config['children_per_generation'] - 1 and gen > 0:
+                if config['mutation_strength_method'] != 'flat':
+                    calculate_all_mutation_strengths(config, gen)
+
```

Comment:
A great start on making worker_run_loop simpler!

It's hard to tell from the diff because of Python's whitespace formatting. Was there more changed here besides refactoring it into a separate method?